nixos/modules/services/continuous-integration/woodpecker/server.nix

```diff
@@ -0,0 +1,82 @@
+{
+  config
+  , lib
+  , pkgs
+  , ...
+}:
+
+let
+  cfg = config.services.woodpecker-server;
+in {
+  meta.maintainers = [ lib.maintainers.janik ];
+
+
+  options = {
+    services.woodpecker-server = {
+      enable = lib.mkEnableOption (lib.mdDoc "Woodpecker-Server a CI/CD application for automatic builds, deployments and tests.");
+      package = lib.mkPackageOptionMD pkgs "woodpecker-server" { };
+      stateDir = lib.mkOption {
+        default = "/var/lib/woodpecker";
+        type = lib.types.str;
+        description = lib.mdDoc "woodpecker data directory.";
+      };
+
+      environment = lib.mkOption {
+        default = {};
+        type = lib.types.attrsOf lib.types.str;
+        example = lib.literalExpression
+          ''
+            {
+              WOODPECKER_AGENT_SECRET = "your-shared-secret-goes-here";
+              WOODPECKER_HOST = "https://woodpecker.example.tld";
+              WOODPECKER_OPEN = "true";
+              WOODPECKER_GITEA = "true";
+              WOODPECKER_GITEA_CLIENT = "ffffffff-ffff-ffff-ffff-ffffffffffff";
+              WOODPECKER_GITEA_SECRET = "gto_**************************************";
+              WOODPECKER_GITEA_URL = "https://git.example.tld";
+            }
+          '';
+        description = lib.mdDoc "woodpecker config envrionment variables";
+      };
+      user = lib.mkOption {
+        type = lib.types.str;
+        default = "woodpecker";
+        description = lib.mdDoc "service user.";
+      };
+    };
+  };
+
+  config = lib.mkIf cfg.enable {
+    systemd.tmpfiles.rules = [
+      "d '${cfg.stateDir}' 0750 ${cfg.user} woodpecker - -"
+    ];
+    systemd.services = {
+      woodpecker-server = {
+        description = "Woodpecker-Server Service";
+        wantedBy = [ "multi-user.target" ];
+        after = [ "network-online.target" ];
+        wants = [ "network-online.target" ];
+        serviceConfig = {
+          User = cfg.user;
+          WorkingDirectory = cfg.stateDir;
+          ExecStart = "${pkgs.woodpecker-server}/bin/woodpecker-server";
+          Restart = "on-failure";
+          RestartSec = 120;
+        };
+        environment = {
+        } // cfg.environment;
```

Comment:
```suggestion
        inherit (cfg) environment;
```

Reply:
could you elaborate a bit why you would want to change it? Is there some convention or officially preferred way somewhere?

Reply:
This is the more idiomatic way in nixpkgs, so it's the convention. Also you don't need to merge an empty attrset with environment.